### PR TITLE
set secrets

### PR DESCRIPTION
**Pull request by felixtheflex**
### Context

Set the propensity to be able to manage secrets in a file

### Changes proposed in this pull request

Currenty we have too many secrets on GhA (GitHub Actions). The idea is to consolidate these into a single file where application. The long term objectives is to move this Azure keyVault

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -79,11 +79,9 @@ jobs:
 
 
     - name: Trigger QA Deployment
-      if: ${{ success() && github.ref == 'refs/heads/master' }}
+      if: ${{ success() && github.ref == 'refs/heads/420-store-secrets-as-yml-in-github-secrets' }}
```
Comment (vigneshmsft) [inline .github/workflows/build.yml]: testing changes? file doesn't have to be in this PR.

File: terraform/variables.tf [hunk 2]
```diff
@@ -32,6 +32,8 @@ variable dockerhub_username {}
 
 variable dockerhub_password {}
 
+variable paas_app_secrets_file { default = "./terraform/app_secrets.yml" }
+
 variable paas_settings__basic_auth__username {} #secrets
 variable paas_settings__basic_auth__password {} #secrets
 variable paas_settings__dttp__client_id {}      #secrets
```
Comment (vigneshmsft) [inline terraform/variables.tf:39]: can be removed? as they are moved to the secrets file?

File: terraform/terraform.tf [hunk 3]
```diff
@@ -38,6 +38,7 @@ module paas {
   worker_app_memory           = var.paas_worker_app_memory
   log_url                     = var.paas_log_url
   docker_credentials          = local.docker_credentials
+  app_secrets_variable        = local.app_secrets
 
   settings__basic_auth__username = var.paas_settings__basic_auth__username
   settings__basic_auth__password = var.paas_settings__basic_auth__password
```
Comment (vigneshmsft) [inline terraform/terraform.tf:44]: remove? moved to secrets file?

File: terraform/modules/paas/variables.tf [hunk 4]
```diff
@@ -23,36 +23,32 @@ variable log_url {}
...
-variable settings__dttp__tenant_id {}
-variable settings__dttp__client_secret {} #secrets
-variable settings__dttp__api_base_url {}
+#variable settings__basic_auth__username {}
+#variable settings__basic_auth__password {} #secrets
 
+#variable settings__dttp__client_id {}
+#variable settings__dttp__tenant_id {}
+#variable settings__dttp__client_secret {} #secrets
+#variable settings__dttp__api_base_url {}
```
Comment (vigneshmsft) [inline terraform/modules/paas/variables.tf:32]: these can be removed as well.

**Review by vigneshmsft — APPROVED** [review]

**Review by felixtheflex — COMMENTED** [review]
d

File: terraform/variables.tf (hunk 2 above)
Comment (felixtheflex) [reply terraform/variables.tf]: removed.

File: terraform/terraform.tf (hunk 3 above)
Comment (felixtheflex) [reply terraform/terraform.tf]: resolved..

**Review by defong — APPROVED** [review]

**Pull request merged**